httpx/error.go

```diff
@@ -0,0 +1,64 @@
+package httpx
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/zenledger-io/go-utils/logger"
+	"go.uber.org/zap"
+)
+
+// Error is a JSON-serializable HTTP error.
+type Error struct {
```

Comment:
I didn't add that because this isn't a Go `error` by design. This type is supposed to be a JSON-serializable HTTP error only. I don't want to give the impression that this type is supposed to be returned as an `error` value from a function.

Reply:
Ah go it